utils/slice.go

```diff
@@ -19,6 +23,56 @@ func IndexOf(strs []string, str string) int {
 	return -1
 }
 
+// NormalizeString normalizes a string as NFKC
+func NormalizeString(str string) string {
+	return norm.NFKC.String(str)
+}
+
+// ValidateString checks if the given string is:
+//
+// 1. non-empty
+// 2. entirely composed of utf8 runes
+// 3. normalized as NFKC
+// 4. does not contain any forbidden Unicode code points
+func ValidateString(str string, forbidden string) error {
+	return validateString(str, false, forbidden)
+}
```

Comment:
I think in this `ValidateString` method, we should always forbid `_`. Perhaps, any validation method that needs more flexibility can later export `validateString` if needed to customize what to forbid. I don't like that we need to pass the default separator everywhere (just like if we had to pass a bool everywhere making it error-prone). Basically, `ValidateString` shouldn't be flexible. What do you think?